### PR TITLE
implemented save playlist use case and extended complete playlist ent…

### PR DESCRIPTION
**Pull request by mmahekk**
I implemented the save playlist use case according to the specifications where the user can get a JSON file of the playlist. However, keep in mind that this implementation will need to be updated based on the structure of the JSON file as I was unable to have access to the previous tempjson folder. I also would like someone to review and look over the files I changed to double check that the dependency rule was not violated. Moreover, the DAO and presenter are two files I would like someone to review. I modified the CompletePlaylist constructor along with adding setter/getter methods for encapsulation and to better suit the save playlist use case. Let me know what changes need to be made.

### CODE REVIEW

File: src/entity/CompletePlaylist.java [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
 
+import org.json.JSONArray;
+import org.json.JSONObject;
+
 import java.util.ArrayList;
 
 public class CompletePlaylist extends Playlist {
+    private String name;
+    private ArrayList<Song> songs;
+    private String genre;
```
Comment (finesden33) [inline src/entity/CompletePlaylist.java:11]: these can be safely removed

File: src/entity/CompletePlaylist.java [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
     private String youtubeID;
     private String spotifyID;
 
     public CompletePlaylist(String name, String genre, String youtubeID, String spotifyID) {
         super(name, genre);
+        this.songs = new ArrayList<>();
```
Comment (finesden33) [inline src/entity/CompletePlaylist.java:17]: also this can be removed

File: src/entity/CompletePlaylist.java [hunk 3]
```diff
@@ -47,4 +54,54 @@ public ArrayList<CompleteSong> getCompleteSongs() {
...
+        songs.remove(song);
+    }
+
+    @Override
+    public void setGenre(String genre) {
+        this.genre = genre;
+    }
+
+
+    public JSONObject convertToJSON() {
```
Comment (finesden33) [inline src/entity/CompletePlaylist.java:92]: this method can be removed

**Review by finesden33 — CHANGES_REQUESTED** [review]
The comments suggest the changes

**Comment by mmahekk** [comment]
I added the changes you requested. Please review the updated changes.

**Review by finesden33 — APPROVED** [review]

**Pull request merged**